application/mounts.py

```diff
@@ -42,11 +42,21 @@ def _add_mount(self, monitor, mount):
 		mount_uri = mount.get_root().get_uri()
 		volume = mount.get_volume()
 
+		# check if mount has been added as volume already
+		mount_already_in_volume_list = False
```

Comment:
I think this was a bug in GIO as 14.04 is quite old. I remember fixing the same issue with duplicate mounts as I thought GIO was suppose to work this way. After they fixed this issue in GIO they created a problem for us. If possible, please test this on some newer version just to make sure we are not creating another problem.